src/libraries/System.Console/src/System/TermInfo.DatabaseFactory.cs

```diff
@@ -119,7 +119,9 @@ private static bool TryOpen(string filePath, [NotNullWhen(true)] out SafeFileHan
             {
                 // Read in all of the terminfo data
                 long termInfoLength = RandomAccess.GetLength(fd);
-                const int MaxTermInfoLength = 4096; // according to the term and tic man pages, 4096 is the terminfo file size max
+                // according to the term and tic man pages, 4096 is the terminfo file size max
+                // Fedora includes terminfo files that are slightly larger than 4096.
```

Comment:
I think the limit mentioned in the man pages is the maximum size of a compiled entry, and not an upper limit for the entire file. This code is then setting an artificial boundary.
We hadn't hit it before, so probably we're not likely to hit it again soon when we double it.

Reply:
I don't remember why I did it like this 8 years ago or whatever it was.  Is there any reason to have a hard limit at all, rather than just reading/storing until EOF and growing along the way?

Reply:
I don't think so, these files are read from 'secure' locations (system or envvar).
My preference is to remove the limit check.

Reply:
if this limit was added for "safety", not only is the location trusted as you say, but also is there anything stopping the file growing between the length check and reading it? Or on Unix does opening a fd (with O_RDONLY as here or otherwise) somehow give you a snapshot view?

Reply:
_If_ there were a safety aspect, it'd be more about avoiding OOMs by unbounded memory growth.  But I don't think it's an issue.  We should just read the full file, regardless of length.

Reply:
Right, there's no safety aspect, I just wanted to validate my assumption that there's no way with `open()` to somehow freeze the size of the file in a totally reliable way.

Reply:
Based on our discussion, I've removed the file size check.